feat(cmd): Update commands `fetch`/`menu`/`prepare`

craciunoiuc opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

  Kraftkit follows the same guidelines as the Unikraft Open Source Project.

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

This PR bring kraft menu/fetch up to date as they have been left neglected.
This also removes the unused `kraft prepare`.

This PR duplicated the `pull` function, taken from `build` which will have to be refactored in a later PR.

GitHub-Fixes: #768

craciunoiuc committed:
feat(cmd)!: Bring 'fetch' command up to date

Signed-off-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

craciunoiuc committed:
feat(cmd)!: Update 'menu' command to also pull libraries

Signed-off-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

craciunoiuc committed:
feat(cmd)!: Remove unused 'prepare' command

GitHub-Fixes: #768

Signed-off-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

jake-ciolek commented on cmd/kraft/fetch/fetch.go:
```diff
@@ -5,40 +5,62 @@
+
+			The default behaviour of kraft fetch is to download the dependencies
+			of a unikernel.
+		`, "`"),
```
nit: When I do kraft fetch --help, it results in:

> Fetch the dependencies of a unikernel.
> 
> The default behaviour of kraft fetch is to download the dependencies
> of a unikernel.
> %!(EXTRA string=`)

jake-ciolek commented on cmd/kraft/menu/menu.go:
```diff
@@ -5,42 +5,59 @@
-			Open Unikraft's configuration editor TUI`),
+		Long: heredoc.Docf(`
+			Open Unikraft's configuration editor TUI.
+		`, "`"),
```
See above

craciunoiuc committed:
fix: Restore output writer after calling processtree

This is a temporary workaround for incorporating multiple processes in
a command. After calling processtree the original output writer is lost
so writing will no longer work. To fix this we temporarily save it
beforehand.

A proper fix would ensure in the tui package that this writer is
preserved. Thankfully, this is the only place where it manifests right
now.

Signed-off-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
